### PR TITLE
do not intercept touch events if touch disabled

### PR DESCRIPTION
**Pull request by TimothyZhang**

### CODE REVIEW

**Comment by zilongshanren** [comment]
Hi, @TimothyZhang 
Why should we add this guard at the beginning of `interceptTouchEvent`?
If we don't add this, does it cause any issues?
Thanks.

**Comment by TimothyZhang** [comment]
Sorry I forgot to write a description.

The problem is, if we for some reason disable touch for UIScrollView, it will still respond to touch event if a descendant widiget is clicked and dragged.

File: cocos/ui/UIListView.cpp [hunk 1]
```diff
@@ -504,6 +504,12 @@ void ListView::selectedItemEvent(TouchEventType event)
 void ListView::interceptTouchEvent(TouchEventType event, Widget *sender, Touch* touch)
 {
     ScrollView::interceptTouchEvent(event, sender, touch);
+
```
Comment (zilongshanren) [inline cocos/ui/UIListView.cpp:507]: @TimothyZhang 
Why scrollView doesn't call Layout::interceptTouch to propagate the touch event to its parent?

**Comment by zilongshanren** [comment]
@TimothyZhang 
I think we shouldn't propagate the touch event to its parents when touch is disabled.

And the test logic is also strange:

```
if(!_touchEnabled && !_isInterceptTouch) 
```

The _isInterceptTouch is always true so we should remove it, we only need to determine the _touchEnabled field.

Thoughts?

**Comment by zilongshanren** [comment]
it is fixed at https://github.com/cocos2d/cocos2d-x/pull/13253

**Pull request closed without merging**